CMEPS-interface/CMakeLists.txt

```
@@ -5,18 +5,24 @@
 if(CMAKE_Fortran_COMPILER_ID MATCHES "GNU")
   set(CMAKE_Fortran_FLAGS "-g -fbacktrace -ffree-line-length-none")
   if(${CMAKE_Fortran_COMPILER_VERSION} VERSION_GREATER_EQUAL 10)
-    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fallow-argument-mismatch -fallow-invalid-boz -DCPRGNU")
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fallow-argument-mismatch -fallow-invalid-boz")
+    #Set CPP defintions for the ufs/cdeps_share source component of the cmeps target library
+    set(CDEPS_SHARE_DEFS)
+    list(APPEND CDEPS_SHARE_DEFS "-DCPRGNU")
   endif()
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS}")
   set(CMAKE_Fortran_FLAGS_RELEASE "-O2")
   set(CMAKE_Fortran_FLAGS_DEBUG "-O0 -fcheck=bounds -ffpe-trap=invalid,zero,overflow,underflow" )
   set(CMAKE_Fortran_LINK_FLAGS "")
 elseif(CMAKE_Fortran_COMPILER_ID MATCHES "Intel")
   set(CMAKE_Fortran_FLAGS "-g -traceback")
-  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -O -assume realloc_lhs -DCPRINTEL")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -O -assume realloc_lhs")
   set(CMAKE_Fortran_FLAGS_RELEASE "-O2 -fp-model precise")
   set(CMAKE_Fortran_FLAGS_DEBUG "-O0 -check -check noarg_temp_created -check nopointer -fpe0 -ftrapuv -init=snan,arrays")
   set(CMAKE_Fortran_LINK_FLAGS "")
+  #Set CPP defintions for the ufs/cdeps_share source component of the cmeps target library
+  set(CDEPS_SHARE_DEFS)
+  list(APPEND CDEPS_SHARE_DEFS "-DCPRINTEL")
```

```suggestion
  list(APPEND CDEPS_SHARE_DEFS "CPRINTEL")
```
`-D` is not necessary;`cmake` will ignore the `-D` and add in the makefile during the configuration.

---

thanks for these checks, @aerorahul. all changes have been made.